safeopt/gp_opt.py

```diff
@@ -306,6 +272,50 @@ def use_lipschitz(self, value):
             raise ValueError('Lipschitz constant not defined')
         self._use_lipschitz = value
 
+    @property
+    def parameter_set(self):
+        """Discrete parameter samples for Bayesian optimization."""
+        return self._parameter_set
+
+    @parameter_set.setter
+    def parameter_set(self, parameter_set):
+        self._parameter_set = parameter_set
+
+        # Plotting bounds (min, max value
+        self.bounds = list(zip(np.min(self._parameter_set, axis=0),
+                               np.max(self._parameter_set, axis=0)))
+        self.num_samples = [len(np.unique(self._parameter_set[:, i]))
+                            for i in range(self._parameter_set.shape[1])]
+
+    @property
+    def context_fixed_inputs(self):
```

Comment:
I think this is used by the plotting functions. Could you make sure that moving this doesn't break anything?